File: token-procedure-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__TokenProcedureDescriptor.java
package ${package}.${pluginName};

import se.curity.identityserver.sdk.plugin.descriptor.TokenProcedurePluginDescriptor;
import se.curity.identityserver.sdk.procedure.token.AssertionTokenProcedure;
import se.curity.identityserver.sdk.procedure.token.AssistedTokenProcedure;
import se.curity.identityserver.sdk.procedure.token.AuthorizationCodeTokenProcedure;
import se.curity.identityserver.sdk.procedure.token.AuthorizeCodeTokenProcedure;
import se.curity.identityserver.sdk.procedure.token.AuthorizeImplicitTokenProcedure;
import se.curity.identityserver.sdk.procedure.token.BackchannelAuthenticationTokenProcedure;
import se.curity.identityserver.sdk.procedure.token.ClientCredentialsTokenProcedure;
import se.curity.identityserver.sdk.procedure.token.DeviceAuthorizationTokenProcedure;
import se.curity.identityserver.sdk.procedure.token.DeviceCodeTokenProcedure;
import se.curity.identityserver.sdk.procedure.token.IntrospectionApplicationJwtTokenProcedure;
import se.curity.identityserver.sdk.procedure.token.IntrospectionTokenProcedure;
import se.curity.identityserver.sdk.procedure.token.OpenIdAuthorizeEndpointHybridTokenProcedure;
import se.curity.identityserver.sdk.procedure.token.OpenIdUserInfoTokenProcedure;
import se.curity.identityserver.sdk.procedure.token.RefreshTokenProcedure;
import se.curity.identityserver.sdk.procedure.token.RopcTokenProcedure;
import se.curity.identityserver.sdk.procedure.token.TokenExchangeTokenProcedure;

public final class ${pluginName}TokenProcedureDescriptor implements TokenProcedurePluginDescriptor<${pluginName}TokenProcedureConfig>
{
    @Override
    public Class<? extends RefreshTokenProcedure> getOAuthTokenEndpointRefreshTokenProcedure()
    {
        return ${pluginName}RefreshTokenProcedure.class;
    }

    @Override
    public Class<? extends AuthorizeCodeTokenProcedure> getOAuthAuthorizeEndpointCodeTokenProcedure()
    {
        return ${pluginName}AuthorizeCodeTokenProcedure.class;
    }

    @Override
    public Class<? extends AuthorizeImplicitTokenProcedure> getOAuthAuthorizeEndpointImplicitTokenProcedure()
    {
        return ${pluginName}AuthorizeImplicitTokenProcedure.class;
    }

    @Override
    public Class<? extends DeviceAuthorizationTokenProcedure> getOAuthDeviceAuthorizationTokenProcedure()
    {
        return ${pluginName}DeviceAuthorizationTokenProcedure.class;
    }

    @Override
    public Class<? extends IntrospectionApplicationJwtTokenProcedure> getOAuthIntrospectApplicationJwtTokenProcedure()
    {
        return ${pluginName}IntrospectionApplicationJwtTokenProcedure.class;
    }

    @Override
    public Class<? extends AssertionTokenProcedure> getOAuthTokenEndpointAssertionTokenProcedure()
    {
        return ${pluginName}AssertionTokenProcedure.class;
    }

    @Override
    public Class<? extends AuthorizationCodeTokenProcedure> getOAuthTokenEndpointAuthorizationCodeTokenProcedure()
    {
        return ${pluginName}AuthorizationCodeTokenProcedure.class;
    }

    @Override
    public Class<? extends IntrospectionTokenProcedure> getOAuthIntrospectTokenProcedure()
    {
        return ${pluginName}IntrospectionTokenProcedure.class;
    }

    @Override
    public Class<? extends BackchannelAuthenticationTokenProcedure> getOAuthTokenEndpointBackchannelAuthenticationTokenProcedure()
    {
        return ${pluginName}BackchannelAuthenticationTokenProcedure.class;
    }

    @Override
    public Class<? extends ClientCredentialsTokenProcedure> getOAuthTokenEndpointClientCredentialsTokenProcedure()
    {
        return ${pluginName}ClientCredentialsTokenProcedure.class;
    }

    @Override
    public Class<? extends DeviceCodeTokenProcedure> getOAuthTokenEndpointDeviceCodeTokenProcedure()
    {
        return ${pluginName}DeviceCodeTokenProcedure.class;
    }

    @Override
    public Class<? extends RopcTokenProcedure> getOAuthTokenEndpointRopcTokenProcedure()
    {
        return ${pluginName}RopcTokenProcedure.class;
    }

    @Override
    public Class<? extends TokenExchangeTokenProcedure> getOAuthTokenEndpointTokenExchangeTokenProcedure()
    {
        return ${pluginName}TokenExchangeTokenProcedure.class;
    }

    @Override
    public Class<? extends OpenIdAuthorizeEndpointHybridTokenProcedure> getOpenIdAuthorizeEndpointHybridTokenProcedure()
    {
        return ${pluginName}OpenIdAuthorizeEndpointHybridTokenProcedure.class;
    }

    @Override
    public Class<? extends OpenIdUserInfoTokenProcedure> getOpenIdUserInfoTokenProcedure()
    {
        return ${pluginName}OpenIdUserInfoTokenProcedure.class;
    }

    @Override
    public Class<? extends AssistedTokenProcedure> getOAuthAssistedTokenTokenProcedure()
    {
        return ${pluginName}AssistedTokenProcedure.class;
    }
    
    @Override
    public String getPluginImplementationType()
    {
        return "${pluginName.toLowerCase()}";
    }

    @Override
    public Class<? extends ${pluginName}TokenProcedureConfig> getConfigurationType()
    {
        return ${pluginName}TokenProcedureConfig.class;
    }    
}

File: kotlin-token-procedure-archetype/src/main/resources/archetype-resources/src/main/kotlin/__pluginName__/__pluginName__TokenProcedureDescriptor.kt
package ${package}.${pluginName}

import se.curity.identityserver.sdk.plugin.descriptor.TokenProcedurePluginDescriptor
import se.curity.identityserver.sdk.procedure.token.AssertionTokenProcedure
import se.curity.identityserver.sdk.procedure.token.AssistedTokenProcedure
import se.curity.identityserver.sdk.procedure.token.AuthorizationCodeTokenProcedure
import se.curity.identityserver.sdk.procedure.token.AuthorizeCodeTokenProcedure
import se.curity.identityserver.sdk.procedure.token.AuthorizeImplicitTokenProcedure
import se.curity.identityserver.sdk.procedure.token.BackchannelAuthenticationTokenProcedure
import se.curity.identityserver.sdk.procedure.token.ClientCredentialsTokenProcedure
import se.curity.identityserver.sdk.procedure.token.DeviceAuthorizationTokenProcedure
import se.curity.identityserver.sdk.procedure.token.DeviceCodeTokenProcedure
import se.curity.identityserver.sdk.procedure.token.IntrospectionApplicationJwtTokenProcedure
import se.curity.identityserver.sdk.procedure.token.IntrospectionTokenProcedure
import se.curity.identityserver.sdk.procedure.token.OpenIdAuthorizeEndpointHybridTokenProcedure
import se.curity.identityserver.sdk.procedure.token.OpenIdUserInfoTokenProcedure
import se.curity.identityserver.sdk.procedure.token.RefreshTokenProcedure
import se.curity.identityserver.sdk.procedure.token.RopcTokenProcedure
import se.curity.identityserver.sdk.procedure.token.TokenExchangeTokenProcedure

class ${pluginName}TokenProcedureDescriptor: TokenProcedurePluginDescriptor<${pluginName}TokenProcedureConfig>
{
    override fun getOAuthTokenEndpointRefreshTokenProcedure() = ${pluginName}RefreshTokenProcedure::class.java

    override fun getOAuthAuthorizeEndpointCodeTokenProcedure() = ${pluginName}AuthorizeCodeTokenProcedure::class.java

    override fun getOAuthAuthorizeEndpointImplicitTokenProcedure() = ${pluginName}AuthorizeImplicitTokenProcedure::class.java

    override fun getOAuthDeviceAuthorizationTokenProcedure() = ${pluginName}DeviceAuthorizationTokenProcedure::class.java

    override fun getOAuthIntrospectApplicationJwtTokenProcedure() = ${pluginName}IntrospectionApplicationJwtTokenProcedure::class.java

    override fun getOAuthTokenEndpointAssertionTokenProcedure() = ${pluginName}AssertionTokenProcedure::class.java

    override fun getOAuthTokenEndpointAuthorizationCodeTokenProcedure() = ${pluginName}AuthorizationCodeTokenProcedure::class.java

    override fun getOAuthIntrospectTokenProcedure() = ${pluginName}IntrospectionTokenProcedure::class.java

    override fun getOAuthTokenEndpointBackchannelAuthenticationTokenProcedure() = ${pluginName}BackchannelAuthenticationTokenProcedure::class.java

    override fun getOAuthTokenEndpointClientCredentialsTokenProcedure() = ${pluginName}ClientCredentialsTokenProcedure::class.java

    override fun getOAuthTokenEndpointDeviceCodeTokenProcedure() = ${pluginName}DeviceCodeTokenProcedure::class.java

    override fun getOAuthTokenEndpointRopcTokenProcedure() = ${pluginName}RopcTokenProcedure::class.java

    override fun getOAuthTokenEndpointTokenExchangeTokenProcedure() = ${pluginName}TokenExchangeTokenProcedure::class.java

    override fun getOpenIdAuthorizeEndpointHybridTokenProcedure() = ${pluginName}OpenIdAuthorizeEndpointHybridTokenProcedure::class.java

    override fun getOpenIdUserInfoTokenProcedure() = ${pluginName}OpenIdUserInfoTokenProcedure::class.java

    override fun getOAuthAssistedTokenTokenProcedure() = ${pluginName}AssistedTokenProcedure::class.java
    
    override fun getPluginImplementationType() = "${pluginName.toLowerCase()}"

    override fun getConfigurationType() = ${pluginName}TokenProcedureConfig::class.java    
}
